tasks/main.yml

```diff
@@ -16,8 +17,8 @@
 - name: update
   apt:
     update_cache: true
-    cache_valid_time: "{{ apt_update_cache_valid_time }}"
-  when: apt_update
+    cache_valid_time: "{{ 0 if apt_config_updated is defined and apt_config_updated.changed else apt_update_cache_valid_time }}"
+  when: apt_update or apt_config_updated.changed
```

Comment:
This will fail when `apt_config_updated` is undefined (in case `apt_manage_sources_list` is false)